test/unit_tests/cvode/CXX_serial/cv_test_resize_history.cpp

```diff
@@ -0,0 +1,423 @@
+  // recent time, solution vector, and rhs vector as the first history entry
+  int i_start = hist_size - 2;
+  if (step < hist_size) { i_start = step - 1; }
+
+  for (int i = i_start; i >= 0; i--) { t_hist[i + 1] = t_hist[i]; }
+  t_hist[0] = t_n;
+
+  for (int i = i_start; i >= 0; i--)
+  {
+    N_VScale(one, y_hist[i], y_hist[i + 1]);
+    N_VScale(one, f_hist[i], f_hist[i + 1]);
+  }
+  N_VScale(one, y_n, y_hist[0]);
+  int retval = ode_rhs(t_n, y_n, f_hist[0], user_data);
+  if (retval) { return 1; }
+
+  return 0;
+}
+
+static int resize_history(sunrealtype* t_hist, N_Vector* y_hist, N_Vector* f_hist,
+                          int hist_size, SUNContext sunctx, void* user_data)
+{
+  // Add one new element to the solution and rhs history vectors
+  int new_size = N_VGetLength(y_hist[0]) + 1;
+
+  for (int i = 0; i < hist_size; i++)
+  {
+    N_Vector new_vec      = N_VNew_Serial(new_size, sunctx);
+    sunrealtype* old_data = N_VGetArrayPointer(y_hist[i]);
+    sunrealtype* new_data = N_VGetArrayPointer(new_vec);
+    for (int j = 0; j < new_size; j++) { new_data[j] = old_data[0]; }
+    N_VDestroy(y_hist[i]);
+    y_hist[i] = new_vec;
+  }
+
+  for (int i = 0; i < hist_size; i++)
+  {
+    N_VDestroy(f_hist[i]);
+    f_hist[i]  = N_VClone(y_hist[i]);
+    int retval = ode_rhs(t_hist[i], y_hist[i], f_hist[i], user_data);
+    if (retval) { return 1; }
+  }
+
+  return 0;
+}
+
+static int PrintNordsieckArray(void* cvode_mem, bool print_all)
+{
+  // Access private CVODE memory to print Nordsieck array
+  CVodeMem cv_mem = (CVodeMem)cvode_mem;
+
+  const sunindextype N = N_VGetLength(cv_mem->cv_zn[0]);
+  sunrealtype* vdata   = nullptr;
+
+  cout << setw(4) << "idx" << setw(25) << "zn" << endl;
+
+  int i_max = cv_mem->cv_q;
+  if (print_all) { i_max = cv_mem->cv_qmax; }
+
+  for (int i = 0; i <= i_max; i++)
+  {
+    vdata = N_VGetArrayPointer(cv_mem->cv_zn[i]);
+    for (sunindextype j = 0; j < N; j++)
+    {
+      cout << setw(4) << i << setw(25) << vdata[j] << endl;
+    }
+  }
+  cout << endl;
+
+  return 0;
+}
+
+int main(int argc, char* argv[])
+{
+  // SUNDIALS context object for this simulation
+  sundials::Context sunctx;
+
+  // --------------------
+  // Handle input options
+  // --------------------
+
+  // resize = 0 -- do not resize
+  // resize = 1 -- call resize but with the same problem size
+  // resize = 2 -- grow the problem by one element each time step
+  int resize = 0;
+  if (argc > 1) { resize = atoi(argv[1]); }
+  if (resize > 2)
+  {
+    cerr << "Invalid resize value" << endl;
+    return 1;
+  }
+
+  // Use Adams (1) or BDF methods (2)
+  int method = CV_BDF;
+  if (argc > 2)
+  {
+    if (atoi(argv[2]) == 1) { method = CV_ADAMS; }
+    else if (atoi(argv[2]) == 2) { method = CV_BDF; }
+    else
+    {
+      cerr << "Invalid method option" << endl;
+      return 1;
+    }
+  }
+
+  // Use fixed-point (0) or Newton (1) nonlinear solver
```

Comment:
A bit inconsistent that the first arg is 1 or 2 while this is 0 or 1.

Reply:
The first two are 1 and 2 to match the values of the macros they correspond to